moleculerize/__init__.py

```diff
@@ -0,0 +1,158 @@
+        inventory_groups = {k: v for (k, v) in json_inventory.items() if k != '_meta'}
+    except KeyError:
+        raise RuntimeError('Expected key(s) missing from inventory file! ("_meta", hostvars")')
+
+    for group_name, group_info in inventory_groups.items():
+        if 'children' in group_info and len(group_info['children']) > 0:
+            for child in group_info['children']:
+                if child in inventory_child_groups.keys():
+                    inventory_child_groups[child].add(group_name)
+                else:
+                    inventory_child_groups[child] = {group_name}
+
+    for group_name, group_info in inventory_groups.items():
+        if 'hosts' in group_info.keys():
+            for host in group_info['hosts']:
+                inventory_hosts[host].add(group_name)
+
+                if group_name in inventory_child_groups.keys():
+                    inventory_hosts[host].update(inventory_child_groups[group_name])
+
+    return inventory_hosts
+
+
+def render_molecule_template(inventory_hosts, template_file):
+    """Create a molecule config file from a template.
+
+    Args:
+        inventory_hosts (dict(list(str)): A dictionary of inventory hosts with each host having a list of associated
+            groups.
+        template_file (str): The template file to use for rendering.
+    Returns:
+        str: A molecule config file populated with hosts and groups.
+    """
+
+    template_path = pkg_resources.resource_filename('moleculerize', TEMPLATES_DIR)
+    j2_env = Environment(loader=FileSystemLoader(template_path),
+                         trim_blocks=True,
+                         lstrip_blocks=True,
+                         keep_trailing_newline=True
+                         )
+
+    try:
+        return j2_env.get_template(template_file).render(hosts=inventory_hosts)
+    except TemplateError as e:
+        print(e)
+        raise RuntimeError('Template "{}" not found!'.format(template_file))
+
+
+# ======================================================================================================================
+# Main
+# ======================================================================================================================
+@click.command()
+@click.argument('inv_file', type=click.Path(exists=True))
+@click.option('--template', '-t',
+              type=click.STRING,
+              default=TEMPLATE,
+              help='Molecule config template file')
+@click.option('--output', '-o',
+              type=click.STRING,
+              default=OUTPUT_FILE,
+              help='Output file path for molecule config.')
+def main(inv_file, template, output):
+    """Main program routine.
+
+    Args:
+        inv_file (str): The Ansible inventory file.
+        template (str): The template to use.
+        output (str): The target output file
+    Returns:
+        int: The exit code to return to the shell.
+    """
```

Comment:
When running `moleculerize --help` the docs are read from the doc string which results in unreadable output:

```
Usage: moleculerize [OPTIONS] INV_FILE

  Main program routine.

  Args:     inv_file (str): The Ansible inventory file.     template (str):
  The template to use.     output (str): The target output file Returns:
  int: The exit code to return to the shell.

Options:
  -t, --template TEXT  Molecule config template file
  -o, --output TEXT    Output file path for molecule config.
  --help               Show this message and exit.
```

See `zigzag.cli` for an example on how to format the output to make it pretty.